xcube/core/varexpr.py

```diff
@@ -0,0 +1,270 @@
+# Copyright (c) 2018-2024 by xcube team and contributors
+# Permissions are hereby granted under the terms of the MIT License:
+# https://opensource.org/licenses/MIT.
+
+from typing import Callable, Optional
+
+import numpy as np
+import xarray as xr
+
+from xcube.util.assertions import assert_instance
+
+
+class ExprVar:
+    """A wrapped `xarray.DataArray` to allow safe access in expressions.
+
+    Args:
+        da: The `xarray.DataArray` to be wrapped.
+    """
+
+    def __init__(self, da: xr.DataArray):
+        assert_instance(da, xr.DataArray, name="v")
+        # Note that the double underscore protects access by "name mangling"
+        self.__da = da
+
+    ################################################
+    # Binary operations - comparisons
+
+    def __eq__(self, other):
+        return self.__wrap(self.__da == self.__unwrap(other))
+
+    def __ne__(self, other):
+        return self.__wrap(self.__da != self.__unwrap(other))
+
+    def __le__(self, other):
+        return self.__wrap(self.__da <= self.__unwrap(other))
+
+    def __lt__(self, other):
+        return self.__wrap(self.__da < self.__unwrap(other))
+
+    def __ge__(self, other):
+        return self.__wrap(self.__da >= self.__unwrap(other))
+
+    def __gt__(self, other):
+        return self.__wrap(self.__da > self.__unwrap(other))
+
+    ################################################
+    # Binary operations - emulating numeric type
+
+    def __add__(self, other):
+        return self.__wrap(self.__da + self.__unwrap(other))
+
+    def __radd__(self, other):
```

Comment:
For cases where the right operand is not an `ExprVar`, but the left one is, e.g., `1 + B08`, cannot be dealt with by `__add__` alone, because `self` would be an `int`.